cmd/device-worker/main.go

```diff
@@ -141,6 +142,16 @@ func main() {
 	}
 	configManager := configuration2.NewConfigurationManager(dataDir)
 
+	// Systemd event listener
+	listener := service.NewDBusEventListener()
+	eventCh, err := listener.Connect()
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Start listening to systemd bus events. These events generated from here can start being processed once all observers have been initialized, otherwise
+	// we risk in missing observers whilst processing the events.
+	go listener.Listen()
```

Comment:
I don't think we miss any because the way the events work in systemd is that the first time you tap into the bus you get an event about the service's current status. Subsequent events for that service will be based on a diff from the previous status.

You can try this use case by restarting yggdrasild and you will notice that the workloads are detected as running and the event is propagated to the workload manager once it starts listening.

Reply:
thanks for the clarification. So not a concern at all.